mava/components/jax/building/data_server.py

```diff
@@ -44,6 +44,10 @@ def __init__(
 
     def _create_table_per_trainer(self, builder: SystemBuilder) -> List[reverb.Table]:
         data_tables = []
+        # Default table network config - often overwritten by TrainerInit.
+        if not hasattr(builder.store, "table_network_config"):
+            builder.store.table_network_config = {"table_0": "network_0"}
```

Comment:
Shouldn't we rather create a mock init that creates this config? Instead of having a default in the `data_server`?

Reply:
@DriesSmit So I thought about this. Having a mock init would work for the tests, but I felt that a data server component should be independent of a trainer, to increase flexibility. E.g. maybe we create a new style of trainer that possibly doesn't rely on `table_network_config`. 

Having a default in the data server means if no `table_network_config` is set , it can still work similarly to how it worked before multiple trainers. 

Thoughts on this?

Reply:
I see. I agree that the `data_server` component should be independent of the trainer :+1:  We however need to slightly update the default value as this won't work for agents that don't share weights. Maybe use something like this:

Reply:
You can actually just replace line 49 with an error message telling the user to add a component that creates the table network config.

Reply:
I think I get what you are saying @DriesSmit , but not having a default means you need another component to do something that feels like the `data_server` should be able to handle. 

My thinking - have a reasonable default for `table_network_config`, if other components do trickier stuff , then they overwrite. 

Maybe we can solve this by using a better default or do you think something else?

Reply:
Okay maybe the following will work then. We update line 49 to be:
`builder.store.table_network_config = {"table_0": sort_str_num(builder.store.agent_net_keys.values())}`. If their is a trainer this gets overwritten. Otherwise this default is used. 

Underneath that line we put:
`assert builder.store.network_sampling_setup == enums.NetworkSampler.fixed_agent_networks`. This is just to make sure that the sampler is the correct setting for this assumed default. How does that sound?

Reply:
@DriesSmit Thanks for the pointers. I have made the updates :)